ghstack/submit.py

```diff
@@ -1051,3 +1054,15 @@ def format_url(s: DiffMeta) -> str:
             if noop_pr:
                 print()
                 print("I did NOT close or update PRs previously associated with these commits.")
+
+
+def run_pre_ghstack_hook(sh: ghstack.shell.Shell, base_commit: str, top_commit: str) -> None:
+    """If a `pre-ghstack` git hook is configured, run it."""
+    default_hooks_path = os.path.join(sh.git("rev-parse", "--show-toplevel"), ".git/hooks")
+    hooks_path = sh.git("config", "--default", default_hooks_path, "--get", "core.hooksPath")
+    hook_file = os.path.join(hooks_path, "pre-ghstack")
+
+    if not os.path.isfile(hook_file) or not os.access(hook_file, os.X_OK):
+        return
+
+    sh.sh(hook_file, base_commit, top_commit, stdout=None)
```

Comment:
I think my questions are mostly around whether or not this is the Correct(TM) way to retrieve custom hooks from Git. I'll have to check the Git source code on the matter.

Reply:
I based the approach on the implementation of `git-p4`, which is part of `git` itself: https://github.com/git/git/blob/90d242d36e248acfae0033274b524bfa55a947fd/git-p4.py#L259

I didn't do the bits that are specific for Windows (is `ghstack` supported on windows?), but other than that, this is as official as I could find.

Reply:
Hook implementation in git-p4.py: https://github.com/git/git/blob/89bece5c8c96f0b962cfc89e63f82d603fd60bed/git-p4.py#L224-L288

Reply:
and the C implementation https://github.com/git/git/blob/89bece5c8c96f0b962cfc89e63f82d603fd60bed/hook.c

Reply:
oops I missed your 3hr comment